### PR TITLE
add pytmatrix recipe

### PR DESCRIPTION
**Pull request by kmuehlbauer**
This adds the pytmatrix code to conda-forge.

It has some fortran code inside, which  might be uses in the build process. So things likely break in the first run.

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
ping @jakirkham
I'm in need of a fortran compiler for the three OS to compile one fortran-extension. Q quick search oin conda-forge was giving some issues and PRs but I'm a bit lost. Is there a best practise, on how to do this?

**Comment by jakirkham** [comment]
Use the `gcc` package.

**Comment by kmuehlbauer** [comment]
@jakirkham

Skipping the windows build (fortran problems) for now, to get this into the feedstock. I'll open an issue, when the feedstock is live. Also asking for more maintainers then. This is ready for review/merge from my side, when the CI's are happy!

**Comment by kmuehlbauer** [comment]
ping @gillins 
Although windows build is skipped for the momen: What do I need to get fortran 77 compiler with windows. IIRC this should work, but I surely miss something (appveyor.yaml?).

**Comment by gillins** [comment]
Very good question. Microsoft don't have a fortran compiler and I've lost track of the discussions around switching to mingw and gcc/gfortran.

@jakirkham do you know what the current status is and recommendation for compiling fortran code on Windows? How is scipy currently compiled?

**Comment by patricksnape** [comment]
As far as I know you only have two options on Windows w.r.t a Fortran compiler

  1. Intel Compiler
    - Not free
    - Plays well with Visual Studio
  2. GFortran
    - Free
    - Requires msys2/mingw? I'm never 100% sure what the overlap between the two is.

**Comment by kmuehlbauer** [comment]
@gillins, @patricksnape Thanks for the in-sight and comments. Seems that I have to wait some time for the windows build here. As already said, I'll open an issue after the feedstock is alive.

@ocefpaf, @jakirkham This is ready for transition into feedstock.

File: recipes/pytmatrix/bld.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+"%PYTHON%" setup.py install
+if errorlevel 1 exit 1
```
Comment (patricksnape) [inline recipes/pytmatrix/bld.bat]: Missing terminal new line

Comment (kmuehlbauer) [inline recipes/pytmatrix/bld.bat]: Shouldn't that be catched by linting? Missed that one...

**Comment by kmuehlbauer** [comment]
OK, so we just have the linux and mac builds on board now. Windows needs attention when the fortran issues is settled. Waiting for the CI's. Should finally be ready now.

**Comment by patricksnape** [comment]
OK from me

**Comment by ocefpaf** [comment]
@kmuehlbauer I used `mingwpy` with [some](https://github.com/conda-forge/pyspharm-feedstock/blob/master/recipe/meta.yaml#L15) success on Windows to compile Fortran extensions.

File: recipes/pytmatrix/build.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+$PYTHON setup.py install
```
Comment (ocefpaf) [inline recipes/pytmatrix/build.sh]: Remove this file and use the `script` field.

File: recipes/pytmatrix/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  name: pytmatrix
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/jleinonen/pytmatrix
+  git_tag: {{ version }}
+
+build:
+  number: 0
+
```
Comment (ocefpaf) [inline recipes/pytmatrix/meta.yaml]: Add `script: python setup.py install` here.

File: recipes/pytmatrix/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{% set version = "0.3.0" %}
+
+package:
+  name: pytmatrix
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/jleinonen/pytmatrix
+  git_tag: {{ version }}
```
Comment (ocefpaf) [inline recipes/pytmatrix/meta.yaml]: Change this to use the tarball:
```yaml
fn: pytmatrix-{{ version }}.tar.gz
url: https://github.com/jleinonen/pytmatrix/archive/{{ version }}.tar.gz
sha256: 8733122ab547638fcf586e2069c231e61b411d283d42c4d0923258165f0aaa4b
```

**Comment by ocefpaf** [comment]
Some minor comments and this is ready to go @kmuehlbauer.

**Comment by kmuehlbauer** [comment]
> I used mingwpy with some success on Windows to compile Fortran extensions.

Do those builds also work on 64bit windows? Would be better than nothing.

**Comment by ocefpaf** [comment]
> Do those builds also work on 64bit windows? Would be better than nothing.

Some do some don't. The example I send you above failed on Win64. This [one](https://github.com/conda-forge/gridfill-feedstock/blob/master/recipe/meta.yaml#L12-L17) did work (but note that I had to treat Python 3.5 differently).

**Comment by kmuehlbauer** [comment]
@ocefpaf Why is the windows build not skipped? Hope that we can merge now anyway :grin:

**Comment by jakirkham** [comment]
> @jakirkham do you know what the current status is and recommendation for compiling fortran code on Windows?

Sadly, none at present, @gillins. I expect the suggestion will be to use MinGW-w64. Though I don't know if the MSYS2 wrapped package of this compiler is usable or if needs more work. Related, I'm not sure how we go about using this package. I'm hoping we will talk about this tomorrow.

Though `mingwpy` is an option for Python packages as @ocefpaf. There has been some discussion about whether the strategy taken by `mingwpy` is acceptable as it does some static linking. See some of this discussion here ( https://github.com/conda-forge/openblas-feedstock/issues/2 ).

> How is scipy currently compiled?

We don't currently build it for Windows. This is a mixture of the compiler not being resolved and the fact that we don't have a BLAS for Windows.

Sorry for such sad news, but that is just really the state of things ATM. Honestly we really need to do something for Windows in these areas so it can get back up to speed.

cc @msarahan @mingwandroid

**Comment by gillins** [comment]
Thanks for the update @jakirkham

**Comment by mingwandroid** [comment]
The `m2w64-toolchain` package from the `msys2` channel has `gfortran`. The runtime dependency for `m2w64-toolchain` is `m2w64-gcc-libs`.

So far it's been used to build any of the R packages on the `r` channel that use fortran.

**Comment by mingwandroid** [comment]
.. and hopefully this stuff is within scope for the toolchain meeting!

**Comment by jakirkham** [comment]
Excellent. Sorry to doubt. I just remember we talked about fortran several months back and we weren't sure if that was going to work or not. If it works, that is fantastic.

> .. and hopefully this stuff is within scope for the toolchain meeting!

I pushed for it. 😉 I'll make sure to push again if needed. 😄 Honestly, this seems like a more serious issue to me than Linux compilers.

**Comment by jakirkham** [comment]
@jleinonen, just letting you know that we have added `pytmatrix` to [conda-forge]( http://conda-forge.github.io/ ). This will allow users to easily install this with [`conda`]( http://conda.pydata.org/docs/ ) ( [`conda install -c conda-forge pytmatrix`]( http://anaconda.org/conda-forge/pytmatrix ) ). The maintenance burden here is light as all releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe in [this feedstock (repo)]( https://github.com/conda-forge/pytmatrix-feedstock ). If you have any interest in being added as a maintainer (useful for making new releases for instance), we would be more than happy to add any of you to the maintainers list.

**Pull request merged**